sphinxcontrib/needs/needs.py

```diff
@@ -317,6 +317,7 @@ def setup(app):
     app.connect("doctree-resolved", process_need_outgoing)
     app.connect("doctree-resolved", process_need_count)
     app.connect("doctree-resolved", process_need_func)
+    app.connect("build-finished", delete_temp_dir)
```

Comment:
I think we also need a ``create_temp_dir`` and connect to an early event.
I know it gets created also inside ``needpie.py`` line 236, but I think we should be safe here and create this central folder at a central place. :)